conda/deprecations.py

```diff
@@ -350,22 +374,76 @@ def _get_module(self, stack: int) -> tuple[ModuleType, str]:
 
         raise DeprecatedError("unable to determine the calling module")
 
+    def _generate_user_message(
+        self: Self,
+        deprecate_in: str,
+        remove_in: str,
+        prefix: str,
+        addendum: str | None,
+    ) -> tuple[UserDeprecationType | None, str]:
+        """Generate the standardized deprecation message for users and determine
+        whether the deprecation is pending, active, or past.
+
+        :param deprecate_in: Version in which code will be marked as deprecated.
+        :param remove_in: Version in which code is expected to be removed.
+        :param prefix: The message prefix, usually the function name.
+        :param addendum: Additional messaging. Useful to indicate what to do instead.
+        """
+        return self._generate_message(
+            deprecate_in,
+            remove_in,
+            prefix,
+            addendum,
+            PendingDeprecationWarning=PendingDeprecationWarning,
+            DeprecationWarning=FutureWarning,
+        )
+
+    def _generate_developer_message(
+        self: Self,
+        deprecate_in: str,
+        remove_in: str,
+        prefix: str,
+        addendum: str | None,
+    ) -> tuple[DevDeprecationType | None, str]:
+        """Generate the standardized deprecation message for developers and determine
+        whether the deprecation is pending, active, or past.
+
+        :param deprecate_in: Version in which code will be marked as deprecated.
+        :param remove_in: Version in which code is expected to be removed.
+        :param prefix: The message prefix, usually the function name.
+        :param addendum: Additional messaging. Useful to indicate what to do instead.
+        """
+        return self._generate_message(
+            deprecate_in,
+            remove_in,
+            prefix,
+            addendum,
+            PendingDeprecationWarning=PendingDeprecationWarning,
+            DeprecationWarning=DeprecationWarning,
+        )
```

Comment:
These two seem a little repetitive, is this needed because of the different return type?

Reply:
This is repetitive because in `_get_user_messages` we use `FutureWarning` instead of `DeprecationWarning`